### PR TITLE
Do not truncate the sortable_title index.

### LINKED ISSUE #3690: Sortable title cuts the title without a valid reason
In Plone 5.2 and also in 6 the sortable_title index is a FieldIndex in the catalog used to sort contents based on titles. But the sortable_title method in portal_catalog just cuts the title in the middle if the title is longer that 40 chars.

Master:
https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/CatalogTool.py#L165-L184

https://github.com/plone/Products.CMFPlone/blob/5.2.x/Products/CMFPlone/CatalogTool.py#L190-L211

Is this behavior intended? And if so why?

### PR DESCRIPTION
**Pull request by erral**
Fixes #3690

### CODE REVIEW

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

**Review by ale-rt — APPROVED** [review]
This looks good to me but before merging this one in the RC phase I would like to hear the opinion of the release managers and the Plone FWT

CC @plone/release-team  @plone/framework-team

**Comment by wesleybl** [comment]
It would be nice to have an upgrade step to update the catalog.

**Comment by jensens** [comment]
be careful with catalog upgrade steps. this can take ages on large databases.

File: Products/CMFPlone/tests/testCatalogTool.py
Comment (mauritsvanrees) [inline Products/CMFPlone/tests/testCatalogTool.py:552]: In the new situation a title like this will only confuse, because it will no longer be ignored.  Maybe something like: "Long titles used to be truncated, but we changed this, see issue 3690", in two variants.

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/tests/testCatalogTool.py
Comment (jensens) [reply Products/CMFPlone/tests/testCatalogTool.py:552]: @erral any chance to change this and merge it?

Comment (erral) [reply Products/CMFPlone/tests/testCatalogTool.py]: I have just changed the titles in the tests.

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

**Comment by wesleybl** [comment]
@erral do you plan on doing a step upgrade for this?

**Comment by jensens** [comment]
Dange-zone: upgrade steps on the catalog for databases with many entries may take ages.

**Comment by erral** [comment]
> .



> Danger-zone: upgrade steps on the catalog for databases with many entries may take ages.

So what do you think, should I do one?

**Comment by mauritsvanrees** [comment]
How about running the original code on `brain.Title`?  Then only if the title is longer than the previous maximum, we get the object and reindex it.
Something like this (untested):

```
for brain in catalog.getAllBrains():
    title = brain.Title
    sortabletitle = mapUnicode(safe_text(title)).lower().strip()
    sortabletitle = num_sort_regex.sub(zero_fill, sortabletitle)
    if len(sortabletitle) <= MAX_SORTABLE_TITLE:
        continue
    try:
        obj = brain.getObject()
    except Exception:
        logger.warning("Failed getting object at %s", brain.getPath()
        continue
    brain.reindexObject(idxs="sortable_title", update_metadata=False)
```

**Pull request merged**
